projects/batfish-client/test/org/batfish/client/ClientTest.java

```diff
@@ -0,0 +1,288 @@
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.BOOLEAN);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("\"true\"",
+            invalidBooleanNode.getNodeType(), variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(invalidBooleanNode, variable);
+   }
+
+   @Test
+   public void testValidBooleanValue() throws IOException {
+      JsonNode booleanNode = mapper.readTree("true");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.BOOLEAN);
+      try {
+         Client.validateType(booleanNode, variable);
+      } catch (Exception e) {
+         throw new BatfishException("Unexpected exception");
+      }
+   }
+
+   @Test
+   public void testInvalidComparatorValue() throws IOException {
+      JsonNode invalidComparator = mapper.readTree("\"=>\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.COMPARATOR);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("\"=>\"",
+            invalidComparator.getNodeType(), variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(invalidComparator, variable);
+   }
+
+   @Test
+   public void testValidComparatorValue() throws IOException {
+      JsonNode comparatorNode = mapper.readTree("\">=\"");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.COMPARATOR);
+      try {
+         Client.validateType(comparatorNode, variable);
+      } catch (Exception e) {
+         throw new BatfishException("Unexpected exception");
+      }
+   }
+
+   @Test
+   public void testInvalidIntegerValue() throws IOException {
+      JsonNode invalidInteger = mapper.readTree("1.5");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.INTEGER);
+      thrown.expect(BatfishException.class);
+      errorMessage = generateErrorMessage("1.5",
+            invalidInteger.getNodeType(), variable.getType());
+      thrown.expectMessage(errorMessage);
+      Client.validateType(invalidInteger, variable);
+   }
+
+   @Test
+   public void testValidIntegerValue() throws IOException {
+      JsonNode comparatorNode = mapper.readTree("15");
+      Question.InstanceData.Variable variable
+            = new Question.InstanceData.Variable();
+      variable.setType(Question.InstanceData.Variable.Type.INTEGER);
+      try {
+         Client.validateType(comparatorNode, variable);
+      } catch (Exception e) {
+         throw new BatfishException("Unexpected exception");
+      }
```

Comment:
what is this try .. catch construct in lines 134..138 doing? Why not just have line 135?